node/tools/src/k8s.rs

```diff
@@ -359,3 +354,23 @@ fn get_cli_args(peers: &[NodeAddr]) -> Vec<String> {
         .to_vec()
     }
 }
+
+async fn retry<T, Fut, F>(retries: usize, delay: usize, mut f: F) -> anyhow::Result<T>
+where
+    F: FnMut() -> Fut,
+    Fut: std::future::Future<Output = anyhow::Result<T>>,
+{
+    let delay = std::time::Duration::from_millis(delay.try_into()?);
+    let mut count = 0;
+    loop {
```

Comment:
for count in 0.. {
   ...
}

Reply:
This would have been really clean
```
{
    let mut interval = time::interval(delay);
    for count in 0.. {
        interval.tick().await;
        let result = f().await;
        if result.is_ok() || count > retries {
            return result
        }
    }
}
```
but it seems rust compiler can't figure out that it won't have zero iterations:
```
mismatched types
  expected enum `std::result::Result<T, anyhow::Error>`
found unit type `()`
return a value for the case when the loop has zero elements to iterate on, otherwise consider changing the return type to account for that possibility
```

Reply:
just put Ok(()) at the end (or unreachable).